### PR TITLE
Upgrade Conduit SDK to 0.5.0

### LINKED ISSUE #20: Make connector compatible with latest version of Conduit
### Feature description

As the title implies.

### PR DESCRIPTION
**Pull request by hariso**
### Description

Fixes #20 

### Quick checks:

- [x] There is no other [pull request](https://github.com/conduitio-labs/conduit-connector-notion/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: .golangci.goheader.template [hunk 1]
```diff
@@ -1,4 +1,4 @@
-Copyright © {{ YEAR }} Meroxa, Inc.
+Copyright © {{ copyright-year }} Meroxa, Inc.
```
Comment (hariso) [inline .golangci.goheader.template:1]: {{ YEAR }} is the current year, but when a single year is used in a copyright statement, it's actually the start year.

File: .golangci.yml [hunk 2]
```diff
@@ -78,7 +79,6 @@ linters:
     - unconvert
     # - unparam
     - unused
-    - varcheck
```
Comment (hariso) [inline .golangci.yml:81]: Regarding the 3 removed checks: the linter warned they appear not to be maintained anymore and that their functionality can be found in `unused`.

File: source.go [hunk 3]
```diff
@@ -334,12 +337,12 @@ func (s *Source) pageToRecord(ctx context.Context, page *notion.Page, children n
...
 	}
 
-	return sdk.Record{
-		Metadata:  nil,
-		CreatedAt: time.Now(),
-		Key:       sdk.RawData(page.ID),
-		Payload:   payload,
-	}, nil
+	return sdk.Util.Source.NewRecordCreate(
+		nil,
```
Comment (hariso) [inline source.go:341]: The position will be set in `Source.nextPage` once we're sure it's safe to be saved.

File: spec.go [hunk 4]
```diff
@@ -18,14 +18,13 @@ import (
...
-		Name:         "notion",
-		Summary:      "A Conduit connector for Notion.",
-		Description:  "A Conduit connector for Notion.",
-		Version:      "v0.1.0",
-		Author:       "Meroxa, Inc.",
-		SourceParams: map[string]sdk.Parameter{},
+		Name:        "notion",
+		Summary:     "A Conduit connector for Notion.",
+		Description: "A Conduit connector for Notion.",
+		Version:     "v0.4.0",
```
Comment (samirketema) [inline spec.go:27]: is this tracking the latest sdk/conduit version? If so, should we change to v0.5.0?

Comment (hariso) [reply spec.go:27]: @samirketema this is actually the version of the connector itself. A connector version itself doesn't necessarily need to track the SDK version (i.e. they are released completely separately)

Comment (samirketema) [reply spec.go:27]: ah, yes! I didn't realize I forgot to update the version here when I bumped to v0.3.0, so it threw me off to see `v0.1.0` currently. sounds good!

**Review by samirketema — APPROVED** [review]
looks great, thanks for the speedy update!

**Comment by samirketema** [comment]
FYI, just manually tested this and confirm it works. Will merge and cut a new release. Thanks again @hariso for the updates!

**Comment by hariso** [comment]
> FYI, just manually tested this and confirm it works. Will merge and cut a new release. Thanks again @hariso for the updates!

Thank you for testing it and releasing it!

**Pull request merged**